extension/script/backend/master/mgr.lua

```diff
@@ -166,6 +166,8 @@ function mgr.exitWorker(w)
     for WorkerIdent, threadId in pairs(threadCatalog) do
         if threadId == w then
             threadCatalog[WorkerIdent] = nil
+            local workerChannel = ('DbgWorker(%s)'):format(WorkerIdent)
+            channel.destroy(workerChannel)
```

Comment:
The worker channel is created in the worker thread, so it should also be destroyed in the worker thread.

Reply:
That is what makes sense. The right place would probably be right after `sendToMaster 'exitWorker' {}` in here:

https://github.com/actboy168/lua-debug/blob/a37fecfbd9a1094c668ba7fa7fc0c6418fdaf22a/extension/script/backend/worker.lua#L830-L832

However, channels are non-blocking so if I destroy the channel right after pushing a message, the master may not get the message.

So it is ugly, but that is why I destroy the worker channel in the master. It is a cheap solution.

The clean but complicated option would be to wait for an "exitWorkerAcknowledgement" message from the master.

Please let me know your thoughts.

Reply:
The channel has reference counting. Destroying it only makes it undiscoverable through query, and does not affect references that have already been obtained from query.

Reply:
ok, I see.
Fixed.